cfy_manager/components/postgresql_server/postgresql_server.py

```diff
@@ -377,6 +386,257 @@ def _configure_cluster(self):
         systemd.enable('patroni', append_prefix=False)
         systemd.start('patroni', append_prefix=False)
 
+    def _get_cluster_addresses(self):
+        master = None
+        replicas = []
+        if DATABASE_SERVICE in config[SERVICES_TO_INSTALL]:
+            etcd_cluster_health = common.run(
+                [
+                    'etcdctl',
+                    '--endpoint', 'https://127.0.0.1:2379',
+                    '--ca-file', ETCD_CA_PATH,
+                    'cluster-health',
+                ],
+                ignore_failures=True
+            ).aggr_stdout
+            if (
+                'cluster is unavailable' in etcd_cluster_health
+                or 'failed to list members' in etcd_cluster_health
+            ):
+                raise DBNodeListError(
+                    'Etcd is not responding on this node. '
+                    'Please retry this command on another DB cluster node.'
+                )
+
+            nodes = json.loads(common.sudo([
+                '/opt/patroni/bin/patronictl', '-c', '/etc/patroni.conf',
+                'list', '--format', 'json'
+            ]).aggr_stdout.strip())
+            for node in nodes:
+                server_name = node['Host']
+                if node['Role'] == 'Leader':
+                    master = server_name
+                else:
+                    replicas.append(server_name)
+        elif MANAGER_SERVICE in config[SERVICES_TO_INSTALL]:
+            backends = common.get_haproxy_servers(self.logger)
+            for backend in backends:
+                # svname will be in the form postgresql_192.0.2.48_5432
+                server_name = backend['svname'].split('_')[1]
+                if backend['status'] == 'UP':
+                    master = server_name
+                else:
+                    replicas.append(server_name)
+        else:
+            raise DBNodeListError(
+                'Can only list DB nodes from a manager or DB node.'
+            )
+        return master, replicas
+
+    def _get_node_status(self, address, target_type):
+        url = {
+            'etcd': 'https://{address}:2379/v2/stats/self',
+            'DB': 'https://{address}:8008',
+        }[target_type].format(address=address)
+
+        dead_node_exceptions = (
+            requests.exceptions.ConnectionError,
+            requests.exceptions.Timeout,
+        )
+
+        if DATABASE_SERVICE in config[SERVICES_TO_INSTALL]:
+            # Using the etcd CA cert as it's the same, and the permissions
```

Comment:
# Using the etcd CA cert as it's the same, and the permissions
            # to its directory are more permissive than postgres'

It does explain- the postgres directory is inaccessible (permissions not as permissive)

Reply:
just saying "those are more permissive" doesn't explain it (because the other ones can still be permissive enough, even though these ones are more permissive!), saying "those other aren't permissive enough" would

Reply:
Fair point, added a note that the postgres ones don't even allow access to the CA cert.